### PR TITLE
3.3.5 Don't allow to summon dead hunter pets

### LINKED ISSUE #19770: [3.3.5] Core/Pet: Action bar
**Description:**

If you summon a dead hunter pet you wont get the pet action bar.

**Current behaviour:**

No action bar.

**Expected behaviour:**

You should get an action bar.

**Steps to reproduce the problem:**

1. Kill your pet
2. Run out of range / wait for despawn
3. Summon it

**Branch(es):**

3.3.5

**TC rev. hash/commit:** 

TrinityCore rev. edd2c1c03995 2017-05-17 08:08:46 +0200 (3.3.5 branch) (Win64, Debug, Static) (worldserver-daemon) 

**TDB version:**  CHANGEME Version of the TrinityCore database

**Operating system:** 
Win

### PR DESCRIPTION
**Pull request by jackpoz**
**Changes proposed:**

-  Don't allow to summon dead hunter pets when using "Call pet"

**Target branch(es):** 3.3.5/master

- [x] 3.3.5
- [ ] master

**Issues addressed:** Closes #  (insert issue tracker number)
Closes #19770

**Tests performed:** (Does it build, tested in-game, etc.)
Tried to call a dead hunter pet and got the correct error message.
Revived it, it summoned the pet and revived it.

**Known issues and TODO list:** (add/remove lines as needed)

- [ ] Handle stabled pets checks

### CODE REVIEW

**Comment by malcrom** [comment]
Prevent summoning of dead hunter pet

**Comment by jackpoz** [comment]
> Prevent summoning of dead hunter pet

?

**Comment by illfated** [comment]
I think he meant to suggest a PR title improvement, in terms of English grammar logic (don't allow ≈ prevent) for easier / more immediate interpretation.

**Comment by jackpoz** [comment]
Cleaned up the code the a bit (at the cost of the pet name sometimes appearing and disappearing when trying to summon a dead pet, as the pet is added to world and then removed right away)

File: src/server/game/Spells/SpellEffects.cpp [hunk 1]
```diff
@@ -2977,7 +2977,7 @@ void Spell::EffectSummonPet(SpellEffIndex effIndex)
 
     float x, y, z;
     owner->GetClosePoint(x, y, z, owner->GetCombatReach());
-    Pet* pet = owner->SummonPet(petentry, x, y, z, owner->GetOrientation(), SUMMON_PET, 0);
+    Pet* pet = owner->SummonPet(petentry, x, y, z, owner->GetOrientation(), SUMMON_PET, 0, true);
```
Comment (jackpoz) [inline src/server/game/Spells/SpellEffects.cpp]: https://github.com/TrinityCore/TrinityCore/pull/23271/files#diff-b3ff7e4adf5326d9790d99d1dee0da2dR2945 should return an error message about pet being dead too

**Comment by jackpoz** [comment]
I would prefer not to spawn a dead pet and then unsummon it...

**Comment by jackpoz** [comment]
For the time being, this is what we will get as it's still better than the current implementation

**Pull request merged**